HLTrigger/Egamma/plugins/HLTScoutingEgammaProducer.cc

```diff
@@ -212,16 +213,23 @@ void HLTScoutingEgammaProducer::produce(edm::StreamID sid, edm::Event& iEvent, e
     std::vector<DetId> mDetIds = EcalClusterTools::matrixDetId((topology), (*SCseed).seed(), rechitMatrixSize);
 
     int detSize = mDetIds.size();
-    std::vector<uint32_t> mDetIdIds(detSize, 0);
-    std::vector<float> mEnergies(detSize, 0.);
-    std::vector<float> mTimes(detSize, 0.);
+    std::vector<uint32_t> mDetIdIds;
+    std::vector<float> mEnergies;
+    std::vector<float> mTimes;
+    mDetIdIds.reserve(detSize);
+    mEnergies.reserve(detSize);
+    mTimes.reserve(detSize);
 
     for (int i = 0; i < detSize; i++) {
-      mDetIdIds[i] = mDetIds[i];
-      mEnergies[i] =
-          MiniFloatConverter::reduceMantissaToNbitsRounding(recHitE(mDetIds[i], *rechits), mantissaPrecision);
-      if (saveRecHitTiming)
-        mTimes[i] = MiniFloatConverter::reduceMantissaToNbitsRounding(recHitT(mDetIds[i], *rechits), mantissaPrecision);
+      if (not rechitZeroSuppression or recHitE(mDetIds[i], *rechits) > 0) {
+        mDetIdIds.push_back(mDetIds[i]);
+        mEnergies.push_back(
+            MiniFloatConverter::reduceMantissaToNbitsRounding(recHitE(mDetIds[i], *rechits), mantissaPrecision));
```

Comment:
```suggestion
      auto const recHit_en = recHitE(mDetIds[i], *rechits);
      if (not rechitZeroSuppression or recHit_en > 0) {
        mDetIdIds.push_back(mDetIds[i]);
        mEnergies.push_back(
            MiniFloatConverter::reduceMantissaToNbitsRounding(recHit_en, mantissaPrecision));
```
If `rechitZeroSuppression==false`, `recHitE` is calculated once (L227). If `rechitZeroSuppression==true`, `recHitE` is calculated once (L224) or twice (L224 and L227). This should avoid calculating it twice.